### ISSUE TITLE
Language server crash

### ISSUE THREAD

**Issue by BladeMF** [phpactor/phpactor#1735]
This is the output:
```
PHP Fatal error:  Nesting level too deep - recursive dependency? in /root/.shared_extensions/phpactor2/lib/CodeTransform/Adapter/TolerantParser/Refactor/TolerantExtractExpression.php on line 96
Fatal error: Nesting level too deep - recursive dependency? in /root/.shared_extensions/phpactor2/lib/CodeTransform/Adapter/TolerantParser/Refactor/TolerantExtractExpression.php on line 96
[Info  - 12:12:21 PM] Connection to server got closed. Server will restart.
[Error - 12:12:21 PM] Request textDocument/codeAction failed.
```
I cannot reproduce it reliably, but I got it like 10 times in the last couple of hours. Every time different actions trigger it.

**Comment by dantleech** [comment]
potentially fixed in #1740

**Comment by BladeMF** [comment]
We'll probably have confirmation by the end of today. Thank you.
